internal/database/db.go

```diff
@@ -44,14 +45,22 @@ type DBConf struct {
 
 // FileInfo is used by ingest for file metadata (path, size, checksum)
 type FileInfo struct {
-	Checksum string
-	Size     int64
-	Path     string
+	Checksum          hash.Hash
+	Size              int64
+	Path              string
+	DecryptedChecksum hash.Hash
+	DecryptedSize     int64
 }
 
 // Internal variable to ease testing
 var sqlOpen = sql.Open
 
+// hashType returns the identification string for the hash type
+func hashType(h hash.Hash) string {
+	// TODO: Support/check type
```

Comment:
should we add an issue about this, to keep better track of it ?

Reply:
:+1: